pyomo/contrib/alternative_solutions/obbt.py

```diff
@@ -0,0 +1,370 @@
+                condition = results.termination_condition
+            else:
+                try:
+                    results = opt.solve(
+                        model, warmstart=warmstart, tee=tee, load_solutions=False
+                    )
+                except:
+                    results = None
+                if results is None:
+                    results = opt.solve(model, tee=tee, load_solutions=False)
+                condition = results.solver.termination_condition
+            new_constraint = False
+
+            if condition == optimal_tc:
+                if use_appsi:
+                    results.solution_loader.load_vars(solution_number=0)
+                else:
+                    model.solutions.load_from(results)
+                solns.append(Solution(model, all_variables, objective=orig_objective))
+
+                if warmstart:
+                    _add_solution(solutions)
+                obj_val = pe.value(var)
+                variable_bounds[var][idx] = obj_val
+
+                if refine_discrete_bounds and not var.is_continuous():
+                    if sense == pe.minimize and var.lb < obj_val:
+                        bound_name = var.name + "_" + str.lower(bound_dir)
+                        bound = pe.Constraint(expr=var >= obj_val)
+                        setattr(aos_block, bound_name, bound)
+                        new_constraint = True
+
+                    if sense == pe.maximize and var.ub > obj_val:
+                        bound_name = var.name + "_" + str.lower(bound_dir)
+                        bound = pe.Constraint(expr=var <= obj_val)
+                        setattr(aos_block, bound_name, bound)
+                        new_constraint = True
+
+            # An infeasibleOrUnbounded status code will imply the problem is
+            # unbounded since feasibility has been established previously
+            elif condition == infeas_or_unbdd_tc or condition == unbdd_tc:
+                if sense == pe.minimize:
+                    variable_bounds[var][idx] = float("-inf")
+                else:
+                    variable_bounds[var][idx] = float("inf")
+            else:  # pragma: no cover
+                print(
+                    (
+                        "Unexpected condition for the variable {} {} problem."
+                        "TerminationCondition = {}"
+                    ).format(var.name, bound_dir, condition.value)
+                )
+
+            var_value = variable_bounds[var][idx]
+            if not quiet:  # pragma: no cover
+                print(
+                    "Iteration {}/{}: {}_{} = {}".format(
+                        iteration, total_iterations, var.name, bound_dir, var_value
+                    )
+                )
+
+            if idx == 1:
+                variable_bounds[var] = tuple(variable_bounds[var])
+
+            iteration += 1
+
+    # TODO - Remove this block
+    aos_block.deactivate()
+    orig_objective.activate()
```

Comment:
Does this TODO need doing?

Reply:
@jlgearh ???  I'm not sure why you don't just delete aos_block from the model.

Reply:
We will leave the Block (deactivated), so this TODO can be removed.

Reply:
Done